common/src/test/scala/org/neo4j/spark/util/Neo4jImplicitsTest.scala

```diff
@@ -141,4 +144,52 @@ class Neo4jImplicitsTest {
     assertEquals(1, agg.groupByCols().length)
     assertEquals("bar", agg.groupByCols()(0).describe())
   }
+
+  @Test
+  def `should flatten the map`(): Unit = {
+    val input = Map(
+      "foo" -> "bar",
+      "key" -> Map(
+        "innerKey" -> Map("innerKey2" -> "value")
+      )
+    )
+    val expected = Map(
+      "foo" -> "bar",
+      "key.innerKey.innerKey2" -> "value"
+    )
+    val actual = input.flattenMap()
+    Assert.assertEquals(expected, actual)
+  }
+
+  @Test
+  def `should not handle collision`(): Unit = {
+    val input = ListMap(
+      "my" -> Map(
+        "inner" -> Map("key" -> 42424242),
+        "inner.key" -> 424242
+      ),
+      "my.inner" -> Map("key" -> 4242).asJava,
+      "my.inner.key" -> 42
+    )
+    val expected = Map(
+      "my.inner.key" -> 42
+    )
+    val actual = input.flattenMap()
+    Assert.assertEquals(expected, actual)
```

Comment:
(this comment is entirely optional)

the assertion only works because the `ListMap` iterator order follows the declaration order, right?
maybe a more general test would test the flattening of a regular hash map and

1. check the resulting keys
2. check the values as one of the possible candidate (sth like `42 in [42, 4242, 424242, 424242]` in pseudo-code)

Reply:
Yes correct, I wanted to test an ordered use-case to be sure that the behaviour is deterministic, when applicable.